QueueImplementation/VM/CustomCollectionViewModel.cs

```diff
@@ -0,0 +1,363 @@
+﻿using Business.Helper;
+using Business.Interfaces;
+using MvvmCross.ViewModels;
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+
+namespace ViewModel
+{
+	/// <summary>
+	/// Represents the view model used for managing the collection.
+	/// </summary>
+	/// <owner>Anton Petrenko</owner>
+	public sealed class CustomCollectionViewModel<T> : MvxViewModel
+	{
+		/// <summary>
+		/// Holds the add value command.
+		/// </summary>
+		/// <owner>Anton Petrenko</owner>
+		private RelayCommand add;
+
+		/// <summary>
+		/// Holds the all numbers.
+		/// </summary>
+		/// <owner>Anton Petrenko</owner>
+		private ObservableCollection<int> allNumbers;
+
+		/// <summary>
+		/// Holds the collection.
+		/// </summary>
+		/// <owner>Anton Petrenko</owner>
+		private ICustomCollection<int> collection;
+
+		/// <summary>
+		/// Holds the configuration creator.
+		/// </summary>
+		/// <owner>Anton Petrenko</owner>
+		private readonly CollectionConfigCreator<int> configCreator = new CollectionConfigCreator<int>();
```

Comment:
Please move all default values initialization to the constructor.

Reply:
Fixed